### ISSUE TITLE
Enable copy-pasting path into import file picker

### ISSUE THREAD

**Issue by CarePackage17** [NationalSecurityAgency/ghidra#7150]
I often look for binaries with the OS file explorer first, but if I then want to import a file into Ghidra, I have to do the entire navigation again within the file picker window.

The file picker has a sort of address bar, just let me paste into it like I can with Windows Explorer or Nautilus on Ubuntu.

**Comment by dragonmacher** [comment]
You should be able to paste a path into the "File name: " text field and then press Enter.  This will then move the chooser to that directory.

**Comment by CarePackage17** [comment]
Yeah that works, though it's a little confusing from a UX perspective.

Is there anything that precludes the file picker's location textbox working that way (like hardcoded assumptions in the UI framework or something)?

**Comment by dragonmacher** [comment]
> Is there anything that precludes the file picker's location textbox working that way 

Nope.  The file chooser UI was modeled after something in Windows XP.  Hard to believe, but it really hasn't changed much since then.

**Comment by CarePackage17** [comment]
If I wanted to take a stab at making this a thing, where would I find the code?

**Comment by dragonmacher** [comment]
The class is `docking.widgets.filechooser.GFileChooser`.   The code itself is probably not very difficult.  Trickier is dealing with the multithreaded nature of the filechooser.   It can be tricky to get that right.    There are a ton tests that you can run to verify you did't break anything.
